### PR TITLE
feat: return error on attachment import when it already exists and add a flag to override

### PR DESCRIPTION
**Pull request by alexbrdn**
feat: return error on attachment import when it already exists and add a flag to override

### CODE REVIEW

File: internal/app/http/handler.go [hunk 1]
```diff
@@ -365,15 +361,15 @@ func (h *TmcHandler) DeleteTMNameAttachment(w http.ResponseWriter, r *http.Reque
...
 func (h *TmcHandler) PutThingModelAttachmentByName(w http.ResponseWriter, r *http.Request, tmID string, attachmentFileName string, params server.PutThingModelAttachmentByNameParams) {
 	ref := model.NewTMIDAttachmentContainerRef(tmID)
-	h.putAttachment(w, r, convertRepoName(params.Repo), ref, attachmentFileName, r.Header.Get(HeaderContentType))
+	h.putAttachment(w, r, convertRepoName(params.Repo), ref, attachmentFileName, r.Header.Get(HeaderContentType), convertForceParam(params.Force))
 }
 
 func (h *TmcHandler) PutTMNameAttachment(w http.ResponseWriter, r *http.Request, tmName server.TMName, attachmentFileName server.AttachmentFileName, params server.PutTMNameAttachmentParams) {
 	ref := model.NewTMNameAttachmentContainerRef(tmName)
-	h.putAttachment(w, r, convertRepoName(params.Repo), ref, attachmentFileName, r.Header.Get(HeaderContentType))
+	h.putAttachment(w, r, convertRepoName(params.Repo), ref, attachmentFileName, r.Header.Get(HeaderContentType), false)
```
Comment (enricoschuetz) [inline internal/app/http/handler.go:369]: The "force" parameter is always set to false here, so overwriting an attachment will always be rejected. -> params.Force should be passed here.

Should also be covered by a unit test in handler_test.go, that the given request parameter is passed to the called method of the handlerService.

File: internal/app/http/handler.go [hunk 2]
```diff
@@ -365,15 +361,15 @@ func (h *TmcHandler) DeleteTMNameAttachment(w http.ResponseWriter, r *http.Reque
 
 func (h *TmcHandler) PutThingModelAttachmentByName(w http.ResponseWriter, r *http.Request, tmID string, attachmentFileName string, params server.PutThingModelAttachmentByNameParams) {
```
Comment (enricoschuetz) [inline internal/app/http/handler.go:362]: To make it clear and align the naming with putting attachments to a TM-name (PutTMNameAttachment), I suggest to rename it to PutTMIDAttachment.

File: internal/app/cli/copy.go [hunk 3]
```diff
@@ -94,7 +94,7 @@ func Copy(ctx context.Context, repo model.RepoSpec, toRepo model.RepoSpec, searc
 		}
 		if tmCopied { // copy tm name attachments if at least one tm has been copied successfully
```
Comment (enricoschuetz) [inline internal/app/cli/copy.go:95]: We should change the behavior here a bit, cause it feels strange for following szenario:

Given:   copy catalog A to a new catalog B
And Given:  a new attachment added to a TM in catalog A
When:  copy again catalog A to catalog B
Then:  the new attachment in catalog A is not copied to catalog B, as all TM's fail to copy with error "TM exists"

So at the moment it is only possible to get the new attachment copied with `--force`, but this means also that all other TM's and attachments in catalog B are overwritten.

For the scenario above, I would expect the new attachment can be copied without any `--force` and the output would look like this:
```
error   TM xy already exists as xy
error   could not import attachment  README.md: attachment already exists
OK      copied attachment new-attachment.json
```

Comment (alexbrdn) [reply internal/app/cli/copy.go]: done

File: internal/app/http/handler.go (hunk 2 above)
Comment (alexbrdn) [reply internal/app/http/handler.go]: done

File: internal/app/http/handler.go (hunk 1 above)
Comment (alexbrdn) [reply internal/app/http/handler.go]: fixed

**Pull request merged**